### PR TITLE
Implement `checkArgs` for Sign/Send APIs and change field names of `SendTxArgs`

### PR DESCRIPTION
**Pull request by aidan-kwon**
## Proposed changes

- Sync up field names of `SendTxArgs` with fileds of `TxInternalData` 
- Implement `checkArgs` for Sign/Send APIs

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/api_public_transaction_pool_test.go [hunk 1]
```diff
@@ -177,3 +178,147 @@ func testTxTypeSupport_normalCase(t *testing.T, api PublicTransactionPoolAPI, ct
...
+	args = oriArgs
+	if args.FeeRatio != nil {
+		args.FeeRatio = nil
+
+		_, err := api.SignTransaction(ctx, args)
+		assert.Equal(t, "json:\"feeRatio\" is required for "+(*args.TypeInt).String(), err.Error())
+	}
+}
+
+// testTxTypeSupport_noFieldValues tests error handling for not assigned field values.
```
Comment (jimni1222) [inline api/api_public_transaction_pool_test.go]: Can you modify this comments ?

File: api/tx_args.go [hunk 2]
```diff
@@ -108,12 +155,26 @@ func (args *SendTxArgs) checkArgs() error {
...
-	// TODO-Klaytn-TxType Arguments validation will be implemented for each tx type
-	//switch *args.TypeInt {
-	//case types.TxTypeLegacyTransaction:
-	//case types.TxTypeValueTransfer:
-	//case types.TxTypeFeeDelegatedValueTransfer:
-	//}
+	argsType := reflect.TypeOf(*args)
+	argsValue := reflect.ValueOf(*args)
+
+	for i := 0; i < argsType.NumField(); i++ {
```
Comment (jimni1222) [inline api/tx_args.go]: Are those enough to check validation of transaction arguments?

File: api/api_public_transaction_pool_test.go (hunk 1 above)
Comment (aidan-kwon) [reply api/api_public_transaction_pool_test.go]: Thank you

File: api/tx_args.go [hunk 3]
```diff
@@ -53,27 +101,27 @@ type NewTxArgs interface {
...
 	HumanReadable *bool              `json:"humanReadable"`
 
-	AccountKey *hexutil.Bytes `json:"Key"`
+	Key *hexutil.Bytes `json:"key"`
 
 	FeePayer *common.Address `json:"feePayer"`
 	FeeRatio *types.FeeRatio `json:"feeRatio"`
 
-	Signatures types.TxSignaturesJSON `json:"Signatures"`
+	TxSignatures types.TxSignaturesJSON `json:"signatures"`
```
Comment (KimKyungup) [inline api/tx_args.go]: @jimni1222 @nogang Caver can support API with `key`, `signatures` json name?

Comment (jimni1222) [reply api/tx_args.go]: For now in caver-js, sendTransaction works only for legacy transaction.
If user wants to send signed transaction, caver-js will send RLP-encoded transaction string with klay_sendRawTransaction.
Before release this feature, caver needs to support this :)

Comment (aidan-kwon) [reply api/tx_args.go]: `key`, `signatures` fields were introduced a few days ago. Support of caver is not considered yet.

Comment (KimKyungup) [reply api/tx_args.go]: Thanks. :)

**Pull request merged**